### PR TITLE
Execution support for GROUP BY, MIN, MAX, and simple predicates

### PR DESCRIPTION
**Pull request by mattdutson**
No support for AVG yet due to lack of a floating-point type. More extensive testing needed of GROUP BY behavior when there are duplicates in the grouping column.

### CODE REVIEW

File: execution/src/main.rs [hunk 1]
```diff
@@ -1,5 +1,79 @@
 pub mod logical_entities;
```
Comment (johnchronis) [inline execution/src/main.rs]: Did you include this file by accident?
